qlib/Swagger.qm

```diff
@@ -2672,6 +2662,48 @@ public class OtherParameter inherits AbstractParameterObject, SchemaBase {
 
     #! validates the value against the schema definition
     check(bool serialize, bool request, string path, string method, string name, reference value) {
+        # issue #2388 ensure that query parameters are converted to their target type first
+        if (inLoc == "query") {
+            switch (type) {
+                case "integer":
+                    if (value !~ /^[0-9]+$/)
```

Comment:
This regex does not handle negative numbers

Reply:
Or with a sign in general...

Reply:
good point; I'll fix it